src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -786,18 +786,31 @@ pSvg = do
 pCodeWithClass :: PandocMonad m => Text -> Text -> TagParser m Inlines
 pCodeWithClass name class' = try $ do
   TagOpen open attr' <- pSatisfy $ tagOpen (== name) (const True)
-  result <- manyTill pAny (pCloses open)
   let (ids,cs,kvs) = toAttr attr'
       cs'          = class' : cs
-  return . B.codeWith (ids,cs',kvs) .
-    T.unwords . T.lines . innerText $ result
+  code open (ids,cs',kvs)
 
 pCode :: PandocMonad m => TagParser m Inlines
 pCode = try $ do
   (TagOpen open attr') <- pSatisfy $ tagOpen (`elem` ["code","tt"]) (const True)
   let attr = toAttr attr'
-  result <- manyTill pAny (pCloses open)
-  return $ B.codeWith attr $ T.unwords $ T.lines $ innerText result
+  code open attr
+
+code :: PandocMonad m => Text -> Attr -> TagParser m Inlines
+code open attr = do
+  exts <- getOption readerExtensions
+  -- Using `formatCode` in an epub causes certain inline code elements to be
+  -- incorrectly parsed. For example, `<code>epub:switch</code>` will be parsed as
+  -- `Link ("",[],[]) [ Code ("",[],[]) "epub:switch" ]
+  -- ("https://idpf.org/epub/30/spec/epub30-contentdocs.html#sec-xhtml-content-switch","")`
+  -- instead of `Code ("",[],[]) "epub:switch"`.
```

Comment:
Sorry, I'm still baffled by this.
Why would it have this result???

Reply:
I have no idea! Maybe I'm misunderstanding? These are the failing tests:

```
Old
  epub
    features:                                                      FAIL (0.11s)
      
      ------------------------------------------------------------------------
      --- epub/features.native
      +++ /home/elliotbobrow/repos/pandoc/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/test-pandoc/test-pandoc --emulate --quiet epub/features.epub -r epub -w native
      +1508             , Link
      +1509                 ( "" , [] , [] )
      +1510                 [ Code ( "" , [] , [] ) "epub:switch" ]
      +1511                 ( "http://idpf.org/epub/30/spec/epub30-contentdocs.html#sec-xhtml-content-switch"
      +1512                 , ""
      +1513                 )
      -1508             , Code ( "" , [] , [] ) "epub:switch"
      +1623             , Link
      +1624                 ( "" , [] , [] )
      +1625                 [ Code ( "" , [] , [] ) "epub:case" ]
      +1626                 ( "http://idpf.org/epub/30/spec/epub30-contentdocs.html#sec-xhtml-epub-case"
      +1627                 , ""
      +1628                 )
      -1618             , Code ( "" , [] , [] ) "epub:case"
      ------------------------------------------------------------------------
      Use -p '/epub.features/' to rerun this test only.
    formatting:                                                    FAIL (0.23s)
      
      ------------------------------------------------------------------------
      --- epub/formatting.native
      +++ /home/elliotbobrow/repos/pandoc/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/test-pandoc/test-pandoc --emulate --quiet epub/formatting.epub -r epub -w native
      + 608         , Link
      + 609             ( "" , [] , [] )
      + 610             [ Code ( "" , [] , [] ) "CSS Multi-Column Layout" ]
      + 611             ( "http://idpf.org/epub/30/spec/epub30-contentdocs.html#sec-css-multi-column"
      + 612             , ""
      + 613             )
      - 608         , Code ( "" , [] , [] ) "CSS Multi-Column Layout"
      ------------------------------------------------------------------------
      Use -p '/epub.formatting/' to rerun this test only.
```

Reply:
Look at test/epub/features.epub, in content-switch-001.xhtml.
It contains
```html
 <code><a
                                href="http://idpf.org/epub/30/spec/epub30-contentdocs.html#sec-xhtml-content-switch"
                                >epub:switch</a></code> 
```
Your `formatCode` is doing the right thing here, making this a hyperlink with a code element inside. Previously pandoc simply dropped the link, and that's why the test fails. The test should be updated.

Reply:
Ohh, got it. Sorry for all the confusion--I had been having trouble viewing the input to the epub tests. Should be fixed now.